Meadowlark-js/packages/meadowlark-core/src/handler/DocumentReferenceExtractor.ts

```diff
@@ -207,11 +192,11 @@ function foreignKeysFromReferenceGroup(
  * extracts the foreign key (reference) information from the JSON body, in the same form as a
  * natural key: NK#<path1>=<value1>#<path2>=<value2>#<path3>=<value3>#...
  */
-export function extractForeignKeys(entity: TopLevelEntity, body: object): ReferentialConstraint[] {
-  const result: ReferentialConstraint[] = [];
+export function extractDocumentReferences(entity: TopLevelEntity, body: object): DocumentReference[] {
+  const result: DocumentReference[] = [];
 
   (entity.data.meadowlark as EntityMeadowlarkData).apiMapping.referenceGroups.forEach((referenceGroup: ReferenceGroup) => {
-    result.push(...foreignKeysFromReferenceGroup(referenceGroup, body, entity));
+    result.push(...documentReferencesFromReferenceGroup(referenceGroup, body, entity));
   });
```

Comment:
Hard to tell from this diff, but these naming changes plus removing stringification make this extractor easier to understand.